lib/libexfat.c

```diff
@@ -358,26 +358,27 @@ off_t exfat_get_root_entry_offset(struct exfat_blk_dev *bd)
 {
 	struct pbr *bs;
 	int nbytes;
-	unsigned int cluster_size;
+	unsigned int cluster_size, sector_size;
 	off_t root_clu_off;
 
-	bs = (struct pbr *)malloc(sizeof(struct pbr));
+	bs = (struct pbr *)malloc(EXFAT_MAX_SECTOR_SIZE);
```

Comment:
I wonder we should read the boot sector in 4KB. let me check this.

Reply:
I have commented it on your patch.